app/controllers/category_transactions_controller.rb

```diff
@@ -0,0 +1,47 @@
+class CategoryTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :create_new_form, only: %i[new create]
+
+  def new
+  end
+
+  def create
+    respond_to do |format|
+      notice = create_conditions
+      format.html { redirect_to categories_path, notice: notice }
+      format.js { render layout: false }
+    end
+  end
+
+  private
+
+  def create_conditions
+    category_transaction = CategoryTransactions::Creator.new(create_params).call
+    if @form.validate(category_transaction_params) &&
+       category_transaction
+      t('.success')
+    elsif category_transaction == -1
```

Comment:
@sinpai let's use `reform` form for validations. In order not to check such things in controller directly
Notice, that reform's instance has `errors` method, so you can just use them